MBBSEmu.Tests/ExportedModules/Majorbbs/random_Tests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Collections.Generic;
+using Xunit;
+
+namespace MBBSEmu.Tests.ExportedModules.Majorbbs
+{
+    public class random_Tests : ExportedModuleTestBase
+    {
+        private const int GENRDN_ORDINAL = 315;
+        private const int LNGRND_ORDINAL = 390;
+        private const int RAND_ORDINAL = 486;
+
+        [Theory]
+        [InlineData(2, 2)]
+        [InlineData(1, 3)]
+        [InlineData(5, 20)]
+        [InlineData(2500, 10000)]
+        [InlineData(10000, 65000)]
+        [InlineData(1, 6000)]
+        public void genrdnTest(ushort valueMin, ushort valueMax)
+        {
+            Reset();
+
+            ExecuteApiTest(HostProcess.ExportedModules.Majorbbs.Segment, GENRDN_ORDINAL, new List<ushort> { valueMin, valueMax });
+
+            //Verify Results
+            Assert.InRange(mbbsEmuCpuRegisters.AX,valueMin, valueMax);
+        }
+
+        [Fact]
+        public void genrdnTest_invalid()
+        {
+            Reset();
+
+            ExecuteApiTest(HostProcess.ExportedModules.Majorbbs.Segment, GENRDN_ORDINAL, new List<ushort> { 10, 2 });
+
+            var expectedValue = 10;
+
+            //Verify Results
+            Assert.Equal(mbbsEmuCpuRegisters.AX, expectedValue);
+        }
+
+        [Theory]
+        [InlineData(1, 65000)]
+        [InlineData(10000, 65000)]
+        [InlineData(10, 20)]
+        [InlineData(11, 12)]
+        [InlineData(32767, 65500)]
+        public void lngrndTest(ushort valueMin, ushort valueMax)
+        {
+            Reset();
+
+            ExecuteApiTest(HostProcess.ExportedModules.Majorbbs.Segment, LNGRND_ORDINAL, new List<ushort> { (ushort)(valueMin & 0xFFFF), (ushort)(valueMin >> 16), (ushort)(valueMax & 0xFFFF), (ushort)(valueMax >> 16) });
+
+            //Verify Results
+            Assert.InRange(mbbsEmuCpuRegisters.GetLong(), valueMin, valueMax);
+        }
+
+        [Fact]
+        public void lngrndTest_invalid()
+        {
+            Reset();
+
+            var valueMin = 32000;
+            var valueMax = 16000;
+            var expectedValue = 32000;
+
+            ExecuteApiTest(HostProcess.ExportedModules.Majorbbs.Segment, LNGRND_ORDINAL, new List<ushort> { (ushort)(valueMin & 0xFFFF), (ushort)(valueMin >> 16), (ushort)(valueMax & 0xFFFF), (ushort)(valueMax >> 16) });
+
+            //Verify Results
+            Assert.Equal(mbbsEmuCpuRegisters.GetLong(), expectedValue);
+        }
+
+        [Fact]
+        public void randTest()
+        {
+            Reset();
+
+            ExecuteApiTest(HostProcess.ExportedModules.Majorbbs.Segment,RAND_ORDINAL, new List<ushort>());
+
+            //Verify Results
+            Assert.InRange(mbbsEmuCpuRegisters.AX, 1, 32767);
+        }
```

Comment:
There's no real way to test this exactly, since due to the random nature of the test and the fact that we're not controlling the randomness, this test might pass every single time until finally it fails based on whatever random number is returned.

The only way to control this would be to have the test control the `Random` used by the `Majorbbs.cs`. So you'd have to have some sort of test-only method where the test passes the `Random` to use, and then controls the values spit out by the `Random`

Reply:
This is outside the scope of this review, but a good candidate for a follow-on